src/Microsoft.NetCore.Analyzers/UnitTests/Tasks/DoNotCallBeginInvokeOnDelegateTests.cs

```diff
@@ -0,0 +1,656 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpSecurityCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Tasks.DoNotCallBeginInvokeOnDelegate,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+using VerifyVB = Test.Utilities.VisualBasicSecurityCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Tasks.DoNotCallBeginInvokeOnDelegate,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Tasks.UnitTests
+{
+    public class DoNotCallBeginInvokeOnDelegateTests
+    {
+        #region Basic diagnostic tests
+
+        [Fact]
+        public async void BasicBeginInvokeOnAction()
+        {
+            var code = @"
+Imports System
+
+Class C
+    Public Sub M()
+        Dim action = New Action(AddressOf D)
+        action.BeginInvoke(AddressOf Callback, Nothing)
+    End Sub
+
+    Private Sub D()
+        Console.WriteLine(""Test"")
+    End Sub
+
+    Private Sub Callback(ByVal ar As IAsyncResult)
+        Console.WriteLine(ar.ToString())
+    End Sub
+End Class
+";
+            await VerifyVB.VerifyAnalyzerAsync(code, GetVBResultAt(7, 9));
```

Comment:
After the conversion to markup, it would be good to expand these tests:

```cs
await new VerifyVB.Test
{
  ReferenceAssemblies = ReferenceAssemblies.NetStandard.NetStandard10,
  TestCode = code,
}.RunAsync();

await new VerifyVB.Test
{
  ReferenceAssemblies = ReferenceAssemblies.NetStandard.NetStandard20,
  TestCode = code,
}.RunAsync();

await new VerifyVB.Test
{
  ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31,
  TestCode = code,
}.RunAsync();

await new VerifyVB.Test
{
  ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
  TestCode = code,
}.RunAsync();

await new VerifyVB.Test
{
  ReferenceAssemblies = ReferenceAssemblies.NetFramework.NetFramework472.Default,
  TestState =
  {
    Sources = { code },

    // Ignore markup; no diagnostics will be reported for .NET Framework
    MarkupMode = MarkupMode.Ignore,
  },
}.RunAsync();
```